code/game/objects/structures/barricade/barricade.dm

```diff
@@ -466,3 +475,23 @@
 	nailgun.in_chamber = null
 	nailgun.load_into_chamber()
 	return TRUE
+
+// testing making a prop the idea is to regroup all the check that would cancel a repair into this prop..
+// put it after iswelder check... for repair on installed cades...
+
+/obj/structure/barricade/proc/attackby_welder(obj/item/item, mob/user)
+	if(!HAS_TRAIT(item, TRAIT_TOOL_BLOWTORCH))
+		to_chat(user, SPAN_WARNING("You need a stronger blowtorch!"))
+		return
+
+	if(health == maxhealth)
+		to_chat(user, SPAN_WARNING("[src] doesn't need repairs."))
+		return
+
+	if(user.action_busy)
+		to_chat(user, SPAN_WARNING("You are currently doing something else"))
+		return
+
+	if(damage_state == welder_fix && damage_state != null)
+		to_chat(user, SPAN_WARNING("[src] has sustained too much structural damage to be repaired."))
+		return
```

Comment:
I'm not really sure about this. Wouldn't you want to compare `damage_state` being greater than `welder_fix`?
I.E. if it's damaged more than the minimum threshold for repairing?
Secondly can `damage_state` ever equal null anyway?

```
/obj/structure/barricade/proc/update_damage_state()
	var/health_percent = round(health/maxhealth * 100)
	switch(health_percent)
		if(0 to 25) damage_state = BARRICADE_DMG_HEAVY
		if(25 to 50) damage_state = BARRICADE_DMG_MODERATE
		if(50 to 75) damage_state = BARRICADE_DMG_SLIGHT
		if(75 to INFINITY) damage_state = BARRICADE_DMG_NONE
```

Reply:
good eyes.
so since damage_state value go from 0-3 from none to heavy i am going to make a check >= welder_fix
since welder_fix hold the limit from when you can't repair a cade.

i don't think damage_state can be null it's probably just a check to prevent an error?..(didn't made that check i assume stuff are there for a reason...)